### ISSUE TITLE
kubeadm 1.25.0 deployment over yaml  pull error coredns:v1.93

### ISSUE THREAD

**Issue by andybe** [kubernetes/kubernetes#112131]
### What happened?

From a fresh installation (clean)

```
echo "Remove all overlays"
umount -R /var/lib/containers/storage/overlay
rm -rf /var/lib/containers
mkdir -p /var/lib/containers/storage/overlay-images

echo "Remove old cluster information"
rm -rf /etc/kubernetes /var/lib/calico /var/lib/kubelet /var/lib/crio /var/lib/cni /var/lib/rancher /var/lib/etcd
```

```
 [preflight] You can also perform this action in beforehand using 'kubeadm config images pull'
error execution phase preflight: [preflight] Some fatal errors occurred:
        [ERROR ImagePull]: failed to pull image k8s.gcr.io/coredns:v1.9.3: output: E0830 21:08:48.476406  295281 remote_image.go:242] "PullImage from image service failed" err="rpc error: code = Unknown desc = reading manifest v1.9.3 in k8s.gcr.io/coredns: manifest unknown: Failed to fetch \"v1.9.3\" from request \"/v2/coredns/manifests/v1.9.3\"." image="k8s.gcr.io/coredns:v1.9.3"
time="2022-08-30T21:08:48+02:00" level=fatal msg="pulling image: rpc error: code = Unknown desc = reading manifest v1.9.3 in k8s.gcr.io/coredns: manifest unknown: Failed to fetch \"v1.9.3\" from request \"/v2/coredns/manifests/v1.9.3\"."
, error: exit status 1
```

### What did you expect to happen?

kubeadm 1.25.0 and coredns v.1.9.3:

- OK - upgrade plan from 1.24.3/1.24.4  -> kubeadm upgrade v1.25.0
- OK - kubeadm init  ..... (console without yaml)
- NO - kubeadm init --config=kubeadm.yaml

This error not happen with version 1.24.4 and lower.

```
nodeRegistration:
  criSocket: unix:///var/run/crio/crio.sock
  imagePullPolicy: IfNotPresent  (or Always)
```
Ends up with the error above.

With
```
ImagePullPolcy: Never 
```

Ends up in:

```
kube-system   pod/coredns-95db45d46-cgfdm                    0/1     ErrImagePull       0          3m17s
kube-system   pod/coredns-95db45d46-fnsnz                    0/1     ImagePullBackOff   0          3m17s


kubectl -n kube-system describe pod/coredns-95db45d46-cgfdm


  Normal   Pulling           3m10s (x3 over 3m51s)  kubelet            Pulling image "k8s.gcr.io/coredns:v1.9.3"
  Warning  Failed            3m10s (x3 over 3m50s)  kubelet            Failed to pull image "k8s.gcr.io/coredns:v1.9.3": rpc error: code = Unknown desc = reading manifest v1.9.3 in k8s.gcr.io/coredns: manifest unknown: Failed to fetch "v1.9.3" from request "/v2/coredns/manifests/v1.9.3".

```

### How can we reproduce it (as minimally and precisely as possible)?

```
apiVersion: kubeadm.k8s.io/v1beta3
bootstrapTokens:
  - groups:
      - system:bootstrappers:kubeadm:default-node-token
    token: abcdef.0123456789abcdef
    ttl: 24h0m0s
    usages:
      - signing
      - authentication
kind: InitConfiguration
localAPIEndpoint:
  advertiseAddress: 192.168.0.1
  bindPort: 6443
nodeRegistration:
  criSocket: unix:///var/run/crio/crio.sock
  imagePullPolicy: IfNotPresent
  #name: node-master
  taints: null
  kubeletExtraArgs:
    node-ip: 192.168.0.1
---
apiServer:
  timeoutForControlPlane: 4m0s
apiVersion: kubeadm.k8s.io/v1beta3
certificatesDir: /etc/kubernetes/pki
clusterName: kubernetes
controllerManager:
  extraArgs:
    flex-volume-plugin-dir: "/var/lib/kubelet/plugins/volume/exec"
    feature-gates: "MixedProtocolLBService=true,IPv6DualStack=true"
dns: {}
etcd:
  local:
    dataDir: /var/lib/etcd
imageRepository: k8s.gcr.io
kind: ClusterConfiguration
kubernetesVersion: 1.24.4
networking:
  dnsDomain: cluster.local
  podSubnet: 10.85.0.0/16
  serviceSubnet: 10.96.0.0/16
scheduler: {}
---
apiVersion: kubelet.config.k8s.io/v1beta1
authentication:
  anonymous:
    enabled: false
  webhook:
    cacheTTL: 0s
    enabled: true
  x509:
    clientCAFile: /etc/kubernetes/pki/ca.crt
authorization:
  mode: Webhook
  webhook:
    cacheAuthorizedTTL: 0s
    cacheUnauthorizedTTL: 0s
cgroupDriver: systemd
clusterDNS:
  - 10.96.0.10
clusterDomain: cluster.local
cpuManagerReconcilePeriod: 0s
evictionPressureTransitionPeriod: 0s
fileCheckFrequency: 0s
healthzBindAddress: 127.0.0.1
healthzPort: 10248
httpCheckFrequency: 0s
imageMinimumGCAge: 0s
kind: KubeletConfiguration
logging:
  flushFrequency: 0
  options:
    json:
      infoBufferSize: "0"
  verbosity: 0
featureGates:
  MixedProtocolLBService: true
memorySwap: {}
failSwapOn: false
nodeStatusReportFrequency: 0s
nodeStatusUpdateFrequency: 0s
resolvConf: /run/systemd/resolve/resolv.conf
rotateCertificates: true
runtimeRequestTimeout: 0s
shutdownGracePeriod: 0s
shutdownGracePeriodCriticalPods: 0s
staticPodPath: /etc/kubernetes/manifests
streamingConnectionIdleTimeout: 0s
syncFrequency: 0s
volumeStatsAggPeriod: 0s
```

### Kubernetes version

v.1.25.0

### Cloud provider

kvm

### OS version

 5.15.63-1-lts #1 SMP Thu, 25 Aug 2022 13:40:16 +0000 x86_64 GNU/Linux

### Install tools

Manual personal script
(https://dl.k8s.io/release/$VERSION/bin/linux/amd64/kubectl, https://dl.k8s.io/release/$VERSION/bin/linux/amd64/kubeadm, https://dl.k8s.io/release/$VERSION/bin/linux/amd64/kubelet)

```
/usr/local/bin/
crictl  crio  crio-status  crun  helm  kubeadm  kubectl  kubelet  pinns
```

```
/opt/cni/bin/
bandwidth  bridge  calico  calico-ipam  dhcp  firewall  flannel  host-device  host-local  install  ipvlan  loopback  macvlan  portmap  ptp  sbr  static  tuning  vlan  vrf
```

### Container runtime (CRI) and version (if applicable)

 Starting CRI-O, version: 1.25.0

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by kluevandrew** [comment]
Same error for me

**Comment by kluevandrew** [comment]
See https://github.com/kubernetes/kubernetes/issues/99321 it's now `k8s.gcr.io/coredns/coredns:v1.8.0` instead of `k8s.gcr.io/coredns:v1.8.0` and i don't now why

**Comment by kluevandrew** [comment]
Simple workaround 
```bash
  crictl pull k8s.gcr.io/coredns/coredns:v1.9.3
  ctr --namespace=k8s.io image tag k8s.gcr.io/coredns/coredns:v1.9.3 k8s.gcr.io/coredns:v1.9.3
```

**Comment by andybe** [comment]
> 

@kluevandrew Indeed.Demask he Line in the yaml file:
```
#imageRepository: k8s.gcr.io
```

did the trick. Didn't think so because it's the official repo.  In case of that, the packages are never no longer on the same root level.

**Comment by instantlinux** [comment]
> Simple workaround [for CRI runtime]

And for those of us still using `docker-ce`, the workaround syntax is:
```
docker pull k8s.gcr.io/coredns/coredns:v1.9.3
docker tag k8s.gcr.io/coredns/coredns:v1.9.3 k8s.gcr.io/coredns:v1.9.3 
# then kubeadm init
```

**Comment by andybe** [comment]
> > Simple workaround [for CRI runtime]
> 
> And for those of us still using `docker-ce`, the workaround syntax is:
> 
> ```
> docker pull k8s.gcr.io/coredns/coredns:v1.9.3
> docker tag k8s.gcr.io/coredns/coredns:v1.9.3 k8s.gcr.io/coredns:v1.9.3 
> # then kubeadm init
> ```

To speak about pricipals:

Standard Images shoud be tagged on the root level:

This reference is the default:

```
imageRepository: k8s.gcr.io
```
by my kubeadm.yml

**Comment by mdbooth** [comment]
I don't think this is fixed because we've identified a workaround. @andybe could you re-open this, or a maintainer?

**Comment by Noksa** [comment]
Why the issue is closed if it is not fixed yet?

**Comment by alirezaghey** [comment]
This problem still exists as of K8s `v1.25.4`. One thing to note ist that I'm joining `v1.25.4` control planes to an originally `v1.24.8` plane. The workaround for me was using the following:
```
crictl pull k8s.gcr.io/coredns/coredns:v1.9.3
ctr --namespace=k8s.io image tag k8s.gcr.io/coredns/coredns:v1.9.3 k8s.gcr.io/coredns:v1.9.3
```

I'm not sure, if the problem would exists on an originally `v1.25.4` plane, if we tried to add `v1.26.0` control planes.
One thing I'm sure about is that this problem didn't arise, as I created the pure `v1.24` or `v1.25` clusters.
